community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java

```diff
@@ -216,6 +219,7 @@ public void init( SPI spi, Config config )
                         .getOrCreateRelationshipIndex( InternalAutoIndexing.RELATIONSHIP_AUTO_INDEX, null ) ),
                 spi.autoIndexing().relationships() );
         this.indexManager = new IndexManagerImpl( spi::currentStatement, idxProvider, nodeAutoIndexer, relAutoIndexer );
+        this.contextFactorySupplier = lazySingleton( () -> new Neo4jTransactionalContextFactory( spi, locker ) );
```

Comment:
why do we need contextFactory to be supplier at all? why not just create instance of `Neo4jTransactionalContextFactory` like:
```
this.contextFactory = new Neo4jTransactionalContextFactory( spi, locker );
```
and use it?